src/api/core/accounts.rs

```diff
@@ -153,6 +155,14 @@ pub async fn _register(data: JsonUpcase<RegisterData>, mut conn: DbConn) -> Json
     // Make sure we don't leave a lingering invitation.
     Invitation::take(&email, &mut conn).await;
 
+    if let Some(client_kdf_parallelism) = data.KdfParallelism {
+        user.client_kdf_parallelism = Some(client_kdf_parallelism);
+    }
+
+    if let Some(client_kdf_memory) = data.KdfMemory {
+        user.client_kdf_memory = Some(client_kdf_memory);
+    }
```

Comment:
```suggestion
    user.client_kdf_parallelism = client_kdf_parallelism;
    user.client_kdf_memory = client_kdf_memory;
```

If they aren't `Some`, then they must be `None` anyway, so this is simpler. I assume the other KDF fields have a check to avoid overwriting the default value in the user model.

Although I would probably move these down in reorder as `type`, `iter`, `memory`, `parallelism`.

Reply:
I will reorder them but keep them as written. registerData uses `Option<i32>` for all 4, so to keep it consistent I should use the same pattern as the other 2. Or we should change the other 2 to the same "simple" assignment as you suggested.

Reply:
As I said, the `type` and `iter` fields are different because they have default values in the user model, so you want to avoid overwriting them. See the analogous code in upstream:

https://github.com/bitwarden/server/blob/cf669286eda971e060a52b1852ba910df7e34a87/src/Core/Models/Api/Request/Accounts/RegisterRequestModel.cs#L40-L43

Reply:
yep, once again, when I do this, it doesn't compile. But it compiles and works as it is. I have re-tested it.

Reply:
I meant to write
```rust
user.client_kdf_memory = data.KdfMemory;
user.client_kdf_parallelism = data.KdfParallelism;
```
I verified that this builds.

Reply:
It's official. I am an idiot. I actually did that since it didn't compile and was the logical choice. Hmm.... Well... It helps to save the freakin file. hahaha. ;-)